File: e-livraria-web/src/models/Category.ts
import { Book } from "./Book";

export interface ICategory{
    id?: number;
    name?: string;
    books?: Book[];
}

export class Category {
    private _id: number;
    private _name: string;
    private _books: Book[];

    constructor({
        id = 0,
        name = '',
        books = []
    }: ICategory){
        this._id = id;
        this._name = name;
        this._books = books;
    }

    get id(): number{
        return this._id;
    }

    set id(id: number){
        this._id = id;
    }

    get name(): string{
        return this._name;
    }

    set name(name: string){
        this._name = name;
    }

    get books(): Book[]{
        return this._books;
    }

    set books(books: Book[]){
        this._books = books;
    }
}

File: ecommerceLivraria/src/main/java/com/nrabello/ecommerceLivraria/model/Category.java
package com.nrabello.ecommerceLivraria.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "Categorias")
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="cat_id")
    private Long id;
    @Column(name="cat_nome")
    private String name;

    @ManyToMany
    @JoinTable(name = "Livros_Categorias",
            joinColumns = @JoinColumn(name = "cat_id"),
            inverseJoinColumns = @JoinColumn(name = "liv_id"))
    @JsonIgnore
    private Set<Book> books = new HashSet<>();

    public Category() {
    }

    public Category(Long id, String name, Set<Book> books) {
        this.setId(id);
        this.setName(name);
        this.setBooks(books);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Set<Book> getBooks() {
        return books;
    }

    public void setBooks(Set<Book> books) {
        this.books = books;
    }
}
